### ISSUE TITLE
Instance EncodeOpts DefaultDecodeOpts

### ISSUE THREAD

**Issue by FPtje** [haskell-servant/servant-cassava#2]
The minimal example below throws an instance error on compilation. I believe that error is inappropriate, if not at least very confusing. The code creates a simple `/comedy` path, having it return a single joke in CSV format.

``` haskell
type JokeAPI = "comedy" :> Get '[CSV] [Joke]

data Joke = Joke { text :: String, explanation :: String }
    deriving Generic

joke = Joke { text = "Vegetables", explanation = "There is no joke if you have to explain it!"}

server :: Server JokeAPI
server = return [joke]

apiProxy :: Proxy JokeAPI
apiProxy = Proxy

webApplication :: Application
webApplication = serve apiProxy server

runServer :: IO ()
runServer = run 8081 webApplication

instance ToNamedRecord Joke where
  toNamedRecord (Joke j d) = namedRecord ["joke" .= toField j, "description" .= toField d]

instance DefaultOrdered Joke where
```

This is the error it gives:

```
No instance for (EncodeOpts DefaultDecodeOpts)
  arising from a use of ‘serve’
In the expression: serve apiProxy server
In an equation for ‘webApplication’:
    webApplication = serve apiProxy server
```

Note that it requires an instance of **Encode**Opts for Default**Decode**Opts. The reason for this is very simple. From hackage:

``` haskell
type CSV = (CSV', DefaultDecodeOpts)
```

`CSV` is defined as `(CSV', DefaultDecodeOpts)`. When it's used as a Response type, it will try to read the `DefaultDecodeOpts` as _encoding_ options, which fails because there is no instance of `EncodeOpts`.

I suspect that `CSV` is defined as `(CSV', DefaultDecodeOpts)` to deal with the use cases that demand some kind of CSV content to be accepted in the _body_ of a request. Nevertheless, the documentation explicitly encourages using `CSV` in responses.
http://hackage.haskell.org/package/servant-cassava-0.7/docs/Servant-CSV-Cassava.html

An easy fix would be to change that example or create the instance:

``` haskell
instance EncodeOpts DefaultDecodeOpts where
  encodeOpts _ = defaultEncodeOptions
```

This particular instance is rather hacky, though. Perhaps it would be better to have a `DefaultOptions` with instances for both `EncodeOpts` and `DecodeOpts`, letting `CSV` be based on that.

**Comment by jkarni** [comment]
First, thanks for the wonderfully complete issue.

You're right about the cause, and the solution. I'll add a `DefaultOptions` in a moment.

**Comment by jkarni** [comment]
Released to hackage as `servant-cassava` 0.8. Thanks again!

**Comment by FPtje** [comment]
Wow, that was quick! Thanks!
